libraries/app/api.cpp

```diff
@@ -330,9 +330,12 @@ namespace graphene { namespace app {
 
        if(_app.is_plugin_enabled("elasticsearch")) {
           auto es = _app.get_plugin<elasticsearch::elasticsearch_plugin>("elasticsearch");
-          auto _thread = std::make_shared<fc::thread>("elasticsearch");
-          return _thread->async([&](){ return es->get_account_history(account, stop, limit, start); },
-                "thread invoke for method " BOOST_PP_STRINGIZE(method_name)).wait();
+          if(es.get()->get_running_mode() != elasticsearch::mode::only_save) {
+             auto _thread = std::make_shared<fc::thread>("elasticsearch");
```

Comment:
There should be only one thread for handling all ES queries in `_app`. Creating one thread per query is too expensive.

Reply:
Yea, i agree. Please check https://github.com/bitshares/bitshares-core/pull/1725/commits/07de45e51e2f0ebf35fe24cdcd734a4e54d33500

I dont want the thread to be created if it is not going to be used so i declared the member variable in application but the thread is initialized the first time of a request. Next query the `shared_ptr` will not be `null` and by this will not create an additional thread but use the one available.

Not sure if it is the best way to handle it.

Reply:
Good, thanks.